src/devices/Common/Iot/Device/Common/GeographicPosition.cs

```diff
@@ -0,0 +1,371 @@
+        public bool EqualPosition(GeographicPosition position)
+        {
+            if (position == null)
+            {
+                return false;
+            }
+
+            bool ret;
+            if ((Math.Abs((position.Longitude - Longitude)) < ComparisonEpsilon) && (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon))
+            {
+                ret = true;
+            }
+            else
+            {
+                ret = false;
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// Equality comparer. Compares the two positions for equality within about 1cm.
+        /// </summary>
+        /// <param name="obj">The other position</param>
+        /// <returns>True if the two positions are almost identical</returns>
+        public override bool Equals(object? obj)
+        {
+            GeographicPosition? position = obj as GeographicPosition;
+
+            if (position == null)
+            {
+                return false;
+            }
+
+            if (((Math.Abs(position.Longitude - Longitude) < ComparisonEpsilon) &&
+                (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon)) &&
+                (Math.Abs(position.EllipsoidalHeight - EllipsoidalHeight) < ComparisonEpsilon))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Equality comparer. Compares the two positions for equality within about 1cm.
+        /// </summary>
+        /// <param name="position">The other position</param>
+        /// <returns>True if the two positions are almost identical</returns>
+        public bool Equals(GeographicPosition? position)
+        {
+            if (position == null)
+            {
+                return false;
+            }
+
+            if (((Math.Abs(position.Longitude - Longitude) < ComparisonEpsilon) &&
+                 (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon)) &&
+                (Math.Abs(position.EllipsoidalHeight - EllipsoidalHeight) < ComparisonEpsilon))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns a string representation of this position.
+        /// </summary>
+        /// <returns>A string representation in degrees, minutes and seconds for both latitude and longitude</returns>
+        // TODO: Add different formatting options and add parsing feature
+        public override string ToString()
+        {
+            if (Double.IsNaN(Latitude) || Double.IsNaN(Longitude))
+            {
+                return "NaN";
+            }
+
+            if (Double.IsInfinity(Latitude) || Double.IsInfinity(Longitude))
+            {
+                return "Infinity";
+            }
```

Comment:
is this technically possible? should we be instead not allowing to create such objects?